packages/ui-kit/src/components/PieChart/PieChart.tsx

```diff
@@ -147,26 +148,6 @@ export const PieChartComponent = React.forwardRef<HTMLDivElement, PieChartProps>
 
         const datasets = isDoughnut ? { cutout: '75%' } : { cutout: '0' }
 
-        if (chartRef.current) {
-          const chart = chartRef.current
-
-          chart.data.labels = labels
-          Object.assign(chart.data.datasets[0], {
-            type: variant,
-            data: values,
-            backgroundColor: chartColorPalette,
-            ...datasets
-          })
-
-          chart.options.plugins = {
-            ...chart.options.plugins,
-            ...customPlugins
-          }
```

Comment:
I tried this and it looked pretty the same to me
```
          Object.assign(chart.data.datasets[0], {
            type: variant,
            data: values,
            // backgroundColor: chartColorPalette,
            ...datasets
          })

          // chart.options.plugins = {
          //  ...chart.options.plugins,
          //  ...customPlugins
          // }
```

Reply:
This would disable modifying the `plugins` option, this is important and we are using it in console to remove the tooltip and it's a common use case

Reply:
We have the plugin customization part in the [initial config](https://github.com/propeldata/ui-kit/blob/1a65851fc2d61728fe7432338e3f28cd265ebf29/packages/ui-kit/src/components/PieChart/PieChart.tsx#L175).

I did a few experiments and would like to share the results.

First of all, I tried the following code in StaticPieStory:
```
    chartConfigProps: (config) => {
      const customPlugin = {
        customCanvasBackgroundColor: {
          color: '#ff0000'
        },
      }
      if (config.options) {
        config.options.plugins = {
          ...config.options.plugins,
          ...customPlugin
        }
      }
      return config
    }
```

and it didn't work due to config overriding in [the update part](https://github.com/propeldata/ui-kit/blob/main/packages/ui-kit/src/components/PieChart/PieChart.tsx#L161):
```
          chart.options.plugins = {
            ...chart.options.plugins,
            ...customPlugins
          }
```

When I commented this part, the `chartConfigProps` started working.
<img width="1001" alt="image" src="https://github.com/propeldata/ui-kit/assets/55801864/cf896fc3-b46f-4257-8323-b9465c65ae42">

So, we should remove all styling overrides from the `update` logic.

Secondly, I tried to change the color of the first slice (`['Carnitas', '21250']`) based on the value:
```
    chartConfigProps: (config) => {
      const data = config.data.datasets.at(0)?.data
      const firstItem = data?.at(0)

      const backgroundColor = config.data.datasets.at(0)?.backgroundColor
      if (backgroundColor && firstItem) {
        backgroundColor[0] = firstItem > 30000 ? '#ff0000' : '#102A56'
      }
      config.data.datasets[0].backgroundColor = backgroundColor

      return config
    }
```

Initially, it didn't work.
![pie-update](https://github.com/propeldata/ui-kit/assets/55801864/e0ef329d-c136-4958-85bb-5c54da95b6be)

Then added the following line right before [chart.update()](https://github.com/propeldata/ui-kit/blob/main/packages/ui-kit/src/components/PieChart/PieChart.tsx#L166):
```
          if (chartConfigProps) {
            chartConfigProps(chart.config as ChartConfiguration<'pie' | 'doughnut'>)
          }
```

and it started working!
![pie-update-ok](https://github.com/propeldata/ui-kit/assets/55801864/d0b6b427-a283-474f-95d2-70136407b01a)

The final version of `update` part I use:
```
        if (chartRef.current) {
          const chart = chartRef.current

          chart.data.labels = labels
          Object.assign(chart.data.datasets[0], {
            type: variant,
            data: values,
            ...datasets
          })

          if (chartConfigProps) {
            chartConfigProps(chart.config as ChartConfiguration<'pie' | 'doughnut'>)
          }

          chart.update()

          return
        }
```

These were the only changes I made.

Reply:
This solution looks simpler, chartConfigProps can receive the chart reference and mutate it 👍 

I'm using this change

Reply:
Update: ended up deciding for using the previous approach, this one seems to cause bugs sometimes when passing the datasets as we state in [storybook](https://storybook.propeldata.com/?path=/docs/getting-started-migration-guide--overview), also, re-assigning the chart configuration properties is the recommended way to update charts according to [chart.js docs](https://www.chartjs.org/docs/latest/developers/updates.html)